security/pkg/k8s/chiron/utils.go

```diff
@@ -85,19 +88,42 @@ func GenKeyCertK8sCA(certClient certclient.CertificateSigningRequestInterface, d
 		log.Errorf("CSR generation error (%v)", err)
 		return nil, nil, nil, err
 	}
-
-	// 2. Submit the CSR
 	csrName := getRandomCsrName(secretName, secretNamespace)
+	csrSpec := &cert.CertificateSigningRequestSpec{
+		Request: csrPEM,
+		Groups:  []string{"system:authenticated"},
+		Usages: []cert.KeyUsage{
+			cert.UsageDigitalSignature,
+			cert.UsageKeyEncipherment,
+			cert.UsageServerAuth,
+			cert.UsageClientAuth,
+		},
+	}
+
+	certChain, caCert, err := SignCSRK8sCA(certClient,
+		csrName, csrSpec, dnsName, caFilePath)
+	return certChain, keyPEM, caCert, err
+}
+
+// SignCSRK8sCA generates a certificate from CSR using the K8s CA
+// 1. Submit a CSR
+// 2. Approve a CSR
+// 3. Read the signed certificate
+// 4. Clean up the artifacts (e.g., delete CSR)
+func SignCSRK8sCA(certClient certclient.CertificateSigningRequestInterface,
```

Comment:
Rename to something like GetCertFromK8sCSRAPI? The function is only using the K8s CSR API, not necessarily the K8s CA, right?

Reply:
Can I just change the name to SignCSRK8s? This does not specify who the actual CA is.

Reply:
Would **SignK8sCSR** work? This does not specifically say Kubernates CA will be the signer, just that some CA will sign the K8s CSR.